test/include.test.js

```diff
@@ -579,6 +595,15 @@ describe('include', function() {
           });
         });
 
+        it('does not return included item if hasOne is missing the id property', function(done) {
+          User.findOne({include: {relation: 'profile'}, fields: 'name'}, function(err, user) {
+            should.not.exist(err);
+            should.exist(user);
+            should.not.exist(user.toJSON().profile);
```

Comment:
I'm wondering if we should error out here as well and say that the `id` property is missing. Thoughts?

Reply:
For belongsTo and hasOne, the FK can be null from business function perspective. We can have an option to report as errors if necessary.